examples/AspNetCore/Controllers/WeatherForecastController.cs

```diff
@@ -54,6 +74,9 @@ public IEnumerable<WeatherForecast> Get()
         })
         .ToArray();
 
+        // Optional: Count the freezing days
+        this.freezingDaysCounter.Add(forecast.Count(f => f.TemperatureC < 0));
```

Comment:
People might be confused if they think about the meaning of this counter. Presumably if a web service is issuing weather reports it is probably giving reports for the same days over and over which means this counter will have a value something like num_requested_reports * num_freezing_days_per_report. How about a counter that tracks the number of forecasts instead? That value both seems more plausibly useful and makes more sense to use a Counter to track it.

Reply:
Good idea. We can add forecast as a dimension. 
Lets address in a follow up.

Reply:
I was trying to find a "real" enough use case but didn't want to change the logic of the example. Would you define a forecast here as single invocation of this endpoint? In that case this information is the same as number of requests which one can get from the histogram created by `AddAspNetCoreInstrumentation`.

Reply:
Yeah, each invocation of this method would be Add(1) in my suggestion. I agree it is duplicative most of the time (they might not have those other metrics enabled) but that still seemed like an improvement. If we wanted something that is both reasonable and unique we'd probably need to change the sample at least a little. For example the forecast service could have a cache and the counter counts how many requests weren't found in the cache. Another option could be forecast requests are for a variable number of days and the counter counts how many 10 day forecasts were requested (maybe 10 day forecasts are more expensive to produce than shorter forecasts?).

Reply:
I tried capturing this in a new issue (https://github.com/open-telemetry/opentelemetry-dotnet/issues/4166) so this does not get lost. If you feel extra clarity is needed please add it to the issue.